priv/repo/migrations/20230725063042_create_repo_project_connection.exs

```diff
@@ -0,0 +1,16 @@
+defmodule Lightning.Repo.Migrations.CreateRepoProjectConnection do
+  use Ecto.Migration
+
+  def change do
+    create table(:project_repos, primary_key: false) do
+      add :id, :binary_id, primary_key: true
+      add :github_installation_id, :string
+      add :repo, :string
+      add :branch, :string
+      add :project_id, references(:projects, type: :binary_id)
+      add :user_id, references(:users, type: :binary_id)
```

Comment:
I think we need some kind of `:on_delete` behaviour here... @taylordowns2000 your thoughts?

So right now if a project is about to be deleted (or the user) it won't be possible because of this outstanding relationship.

Do we manage this in the delete/purge logic or add a nillify or delete_all cascade option in the schema?

Reply:
good catch, let me know and I'll add it in

Reply:
@zacck , please add an on_delete cascade for when a project is deleted and an on_delete nillify for when a user is deleted.

in other words: if the user who set the thing up is deleted, the PRC will stay. if the project is deleted, the PRC will also be deleted.

thank you!